### PR TITLE
Move HttpRangeReader out of the spark package

### PR DESCRIPTION
**Pull request by pomadchin**
# Overview

This PR moves HttpRangeReader into the store pacakge.

## Checklist

- [x] [docs/CHANGELOG.rst](https://github.com/locationtech/geotrellis/blob/master/CHANGELOG.md) updated, if necessary

Closes https://github.com/locationtech/geotrellis/issues/3254

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - ArrayTile equals method always returns true if first elements are NaN [#3242](https://github.com/locationtech/geotrellis/issues/3242)
 - Fixed resource issue with JpegDecompressor that was causing a "too many open files in the system" exception on many parallel reads of JPEG compressed GeoTiffs. [#3249](https://github.com/locationtech/geotrellis/pull/3249)
 - Fix MosaicRasterSource, GDALRasterSource and GeoTiffResampleRasterSource behavior [#3252](https://github.com/locationtech/geotrellis/pull/3252)
+- HttpRangeReader should live outside of the Spark package [#3254](https://github.com/locationtech/geotrellis/issues/3254)
```
Comment (CloudNiner) [inline CHANGELOG.md:24]: Noting that this more than a bugfix change and would necessitate a bump of at least the minor version, yes?

Comment (pomadchin) [reply CHANGELOG.md:24]: @CloudNiner do we need to add a `Fixed` section in addition to `Added`?

Comment (CloudNiner) [reply CHANGELOG.md:24]: Eh. I'm no expert but this particular entry still feels like a `Changed`. It's just a change that will cause compiler errors.

Comment (pomadchin) [reply CHANGELOG.md:24]: Yea, it is a breaking change. We can still release it as a minor release but with a note that it breaks bin compat due to an oversight during the 3.0 release.

**Pull request merged**